src/Service/PaymentDetailsService.php

```diff
@@ -146,12 +146,14 @@ public function doPaymentDetails(
         ];
 
         try {
-            $response = $this->checkoutService->paymentsDetails($request);
+            return $this->paymentResponseHandler->handlePaymentResponse(
+                $this->checkoutService->paymentsDetails($request),
+                $orderNumber,
+                $context
+            );
```

Comment:
Just to be a bit more readable

```suggestion
            $result = $this->checkoutService->paymentsDetails($request);
            
            return $this->paymentResponseHandler->handlePaymentResponse(
                $result,
                $orderNumber,
                $context
            );
```